tutorials/telegraf-output-plugin.md

```diff
@@ -0,0 +1,314 @@
+
+Let us now connect to our PostgreSQL instance and introspect the data
+
+```
+$ psql -U postgres -h localhost
+```
+
+The cpu input plugin has one measurement, called cpu, and it's stored in a table of the same name (by default in the public schema).
+So with the SQL query `SELECT * FROM cpu`, depending on how long you left Telegraf running you will see the table populated with some values, and we can find the average usage per cpu core with `SELECT cpu, avg(usage_user) FROM cpu GROUP BY cpu` and the output would look like 
+
+```
+    cpu    |       avg        
+-----------+------------------
+ cpu-total | 8.46385703620795
+ cpu0      | 12.4343351351033
+ cpu1      | 4.88380203380203
+ cpu2      | 12.2718724052057
+ cpu3      | 4.26716970050303
+```
+
+### Adding new Tags or Fields
+
+Your telegraf configuration can change at any moment. An input plugin can be reconfigured to produce different data, or you may decide to index your data with different tags. The plugin we built can dynamically update the created tables with new columns as they appear. The previous configuration we used had no global tags specified other than the `host` tag. We will now add a new global tag in the configuration. Open the file in any text editor and update the `[global_tags]` section (around line 18) with: 
+
+```
+[global_tags]
+  location="New York"
+```
+
+This way all metric collected with the instance of telegraf running with this config will be tagged with location="New York". If we run Telegraf again, collecting the metrics in TimescaleDB
+
+```
+$ telegraf --config telegraf.conf
+```
+
+And after a while we check on the `cpu` table in the database
+
+```
+psql> \dS cpu
+\dS cpu;
+Table "public.cpu"
+      Column      |           Type           
+------------------+--------------------------
+ time             | timestamp with time zone 
+ cpu              | text                     
+ host             | text                     
+ usage_steal      | double precision         
+ usage_iowait     | double precision         
+ usage_guest      | double precision         
+ usage_idle       | double precision         
+ usage_softirq    | double precision         
+ usage_system     | double precision         
+ usage_user       | double precision         
+ usage_irq        | double precision         
+ location         | text                     
+ ```
+
+ The `location` column was added and it contains "New York" for all rows.
+
+ ### Creating a separate metadata table for tags
+
+ The plugin we developed allows the user to select to have the tag sets inserted in a separate
+ table and then referenced via Foreign Keys in the measurement table. This way space can be saved
+ for measurements with low cardinality tag sets. To enable this change, you need to uncomment the `tags_as_foreignkeys` parameter in the plugin config (around line  103 in `telegraf.conf`) and set it to true
```

Comment:
you also save space for high cardinality tags, and there are other benefits to having metadata in a separate table cause you can write certain queries more efficiently